pkg/controller/route/route_controller.go

```diff
@@ -190,9 +194,10 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
 
 					} else {
 						glog.Infof("Created route for node %s %s with hint %s after %v", nodeName, route.DestinationCIDR, nameHint, time.Now().Sub(startTime))
-						return
+						return true, nil
 					}
-				}
+					return false, err
```

Comment:
Returning "false, err" inside of wait.ExponentialBackoff causes wait.ExponentialBackoff to immediately return the error. I do not believe that is the behavior you want. Returning "false, nil" will cause wait.ExponentialBackoff to retry. It is worth noting that if we hit steps unsuccessful retries, this will change the error returned by wait.ExponentialBackoff to a "timed out waiting for the condition" error. However as we are currently ignoring the result of wait.ExponentialBackoff that seems ok.

Reply:
Nice catch!  
And we could record the retry error event to minimize the influence of ```timed out waiting for the condition```.  So people can use ```kubectl get ev``` to see what happend down there.

What do you say?

Reply:
Recording would be an improvement over where we are today, so it sounds good. Just make sure we don't spam either the log or the events stream if we make that change.

Reply:
@aoxn do we need to fix this so that we're not returning `false, err` or that we're returning early when err != nil?

Reply:
also, is there any reason we don't handle the error from `wait.ExponentialBackoff`?